Add license notice to Arm copyrighted source files plus SDPX to already licensed files

harmut01 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->
Apache-2.0 license notice added to source files missing a license header. 
#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->
None
### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->
None
----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [x] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------
@hugueskamba @ARMmbed/mbed-os-maintainers

harrisonmutai-arm committed:
Add license notice to Arm copyrighted source files

Scancode found missing license notices in our source files. This commit
addresses those issues by adding an Apache-2.0 notice to source files
highlighted.

LDong-Arm commented on connectivity/libraries/nanostack-libservice/test/libService/unittest/nsnvmhelper/main.cpp:
```diff
@@ -1,5 +1,18 @@
-/*
- * Copyright (c) 2015 ARM Limited. All rights reserved.
+/* mbed Microcontroller Library
```
```suggestion
/*
```
The first line can be left empty - the text is there in some files for historical reasons only.

0xc0170 replied:
I would also remove this first line as it uses old name that is not correct anymore.

LDong-Arm commented on connectivity/libraries/nanostack-libservice/test/libService/unittest/nsnvmhelper/main.cpp:
```diff
@@ -1,5 +1,18 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2015 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
```
```suggestion
```
We prefer the short header (only Copyright + SPDX-License-Identifier) now? Could you confirm this? @ARMmbed/mbed-os-maintainers

0xc0170 replied:
the license guide as it was approved by the legal says both should be in the file.

> Begin each source header with your copyright line, the SPDX identifier and the standard header for the license that applies to that single file, if it has one. (See SPDX Specification, Appendix V.)

Although some of the newer files have only SDPX and copyright and were approved. I'll review with @harmut01.

I would leave this as it is as it follows the current guideline and only updates the files that are already in the tree.

LDong-Arm reviewed (commented):
Thanks @harmut01, I've added comment regarding the license header format. You changes are probably okay to merge I guess, let's see what the maintainers say. (Of course, no need to clean up existing notices in other files)

harrisonmutai-arm committed:
Add SPDX license identifier to Arm files

Add license identifier to files which Arm owns the copyright to,
and contain either BSD-3 or Apache-2.0 licenses. This is to address
license errors raised by scancode analysis.

0xc0170 reviewed (commented):
As above, lets remove the first line with old name. The rest looks fine to me.

Thanks @harmut01 !
